aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -326,106 +327,136 @@ func resourceAwsLambdaEventSourceMappingRead(d *schema.ResourceData, meta interf
-		FunctionName:                   aws.String(d.Get("function_name").(string)),
-		Enabled:                        aws.Bool(d.Get("enabled").(bool)),
-		MaximumBatchingWindowInSeconds: aws.Int64(int64(d.Get("maximum_batching_window_in_seconds").(int))),
-	}
+	log.Printf("[INFO] Deleting Lambda Event Source Mapping: %s", d.Id())
 
-	// AWS API will fail if this parameter is set (even as default value) for sqs event source.  Ideally this should be implemented in GO SDK or AWS API itself.
-	eventSourceArn, err := arn.Parse(d.Get("event_source_arn").(string))
-	if err != nil {
-		return fmt.Errorf("Error updating event source mapping: %s", err)
+	input := &lambda.DeleteEventSourceMappingInput{
+		UUID: aws.String(d.Id()),
 	}
 
-	if eventSourceArn.Service != "sqs" {
-		if parallelizationFactor, ok := d.GetOk("parallelization_factor"); ok {
-			params.SetParallelizationFactor(int64(parallelizationFactor.(int)))
-		}
-
-		if maximumRetryAttempts, ok := d.GetOkExists("maximum_retry_attempts"); ok {
-			params.SetMaximumRetryAttempts(int64(maximumRetryAttempts.(int)))
-		}
-
-		params.MaximumRecordAgeInSeconds = aws.Int64(int64(d.Get("maximum_record_age_in_seconds").(int)))
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DeleteEventSourceMapping(input)
 
-		if bisectBatchOnFunctionError, ok := d.GetOk("bisect_batch_on_function_error"); ok {
-			params.SetBisectBatchOnFunctionError(bisectBatchOnFunctionError.(bool))
+		if tfawserr.ErrCodeEquals(err, lambda.ErrCodeResourceNotFoundException) {
+			return nil
 		}
 
-		if vDest, ok := d.GetOk("destination_config"); ok {
-			params.SetDestinationConfig(expandLambdaEventSourceMappingDestinationConfig(vDest.([]interface{})))
+		if tfawserr.ErrCodeEquals(err, lambda.ErrCodeResourceInUseException) {
+			return resource.RetryableError(err)
 		}
-	}
 
-	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
-		_, err := conn.UpdateEventSourceMapping(params)
 		if err != nil {
-			if isAWSErr(err, lambda.ErrCodeInvalidParameterValueException, "") ||
-				isAWSErr(err, lambda.ErrCodeResourceInUseException, "") {
-
-				return resource.RetryableError(err)
-			}
 			return resource.NonRetryableError(err)
 		}
+
 		return nil
 	})
-	if isResourceTimeoutError(err) {
-		_, err = conn.UpdateEventSourceMapping(params)
+
+	if tfresource.TimedOut(err) {
+		_, err = conn.DeleteEventSourceMapping(input)
+	}
+
+	if tfawserr.ErrCodeEquals(err, lambda.ErrCodeResourceNotFoundException) {
+		return nil
 	}
+
 	if err != nil {
-		return fmt.Errorf("error updating Lambda Event Source Mapping (%s): %w", d.Id(), err)
+		return fmt.Errorf("error deleting Lambda Event Source Mapping (%s): %w", d.Id(), err)
 	}
 
-	if _, err := waiter.EventSourceMappingUpdate(conn, d.Id()); err != nil {
-		return fmt.Errorf("error waiting for Lambda Event Source Mapping (%s) to update: %w", d.Id(), err)
+	if _, err := waiter.EventSourceMappingDelete(conn, d.Id()); err != nil {
```

Comment:
Otherwise:

```
=== CONT  TestAccAWSLambdaEventSourceMapping_MSK
    testing_new.go:63: Error running post-test destroy, there may be dangling resources: Lambda Event Source Mapping 25de05f1-8cd0-4df1-92f8-a31f54c99530 still exists
--- FAIL: TestAccAWSLambdaEventSourceMapping_MSK (1453.06s)
```